plugins/modules/smart_proxy.py

```diff
@@ -118,21 +118,21 @@ def main():
     )
 
     with module.api_connection():
+        smart_proxy = module.lookup_entity('entity')
+        new_smart_proxy = module.run()
```

Comment:
Don't you want to have this call in the old place? So the `lifecycle_environments = module.foreman_params.pop('lifecycle_environments', [])` gets this parameter out of the way for the `run()`?

Reply:
Done! :-)

Reply:
I seem to have broken it..

Reply:
Fixed!